onnxruntime/core/graph/contrib_ops/bert_defs.cc

```diff
@@ -1128,6 +1128,16 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
                "2D tensor with shape (max_sequence_length, head_size / 2).",
                "T",
                OpSchema::Optional)
+        .Input(9,
+               "custom_pos_ids",
+               "2D tensor with shape (batch_size, sequence_length).",
+               "tensor(int64)",
+               OpSchema::Optional)
+        .Input(10,
+               "custom_causal_attention_mask",
```

Comment:
Can this be named as `attention_mask` instead to be more generic? The input values will anyways be customized for your scenario.

```suggestion
               "attention_mask",
```

Reply:
Please make the definition generic.

For example, attention_bias with 4D shape (batch_size or 1, num_heads or 1, sequence_length, total_sequence_length). It is fine that you implement (batch_size, 1, sequence_length, total_sequence_length) for now.

Reply:
Oh ok, I will make it 4D and update the parameter description.

Reply:
Updated the name and the shape. Attention mask is now of shape (batch_size or 1, num_heads or 1, sequence_length, total_sequence_length), added support for broadcasting as well.